tests/cupy_tests/testing_tests/test_helper.py

```diff
@@ -393,37 +393,47 @@ def test_correct_success(self, xp, dtype1, dtype2):
             return xp.array(-2, dtype=dtype1)
 
 
-@testing.parameterize(
-    {'xp': numpy},
-    {'xp': cupy},
-)
+@testing.parameterize(*testing.product({
+    'xp': [numpy, cupy],
+    'shape': [(3, 2), (), (3, 0, 2)],
+}))
 @testing.gpu
 class TestShapedRandom(unittest.TestCase):
 
     @testing.for_all_dtypes()
     def test_shape_and_dtype(self, dtype):
-        a = testing.shaped_random((2, 3), self.xp, dtype)
-        self.assertTrue(a.shape == (2, 3))
+        a = testing.shaped_random(self.shape, self.xp, dtype)
+        self.assertIsInstance(a, self.xp.ndarray)
+        self.assertTrue(a.shape == self.shape)
         self.assertTrue(a.dtype == dtype)
 
     @testing.for_all_dtypes(no_bool=True, no_complex=True)
     def test_value_range(self, dtype):
-        a = testing.shaped_random((2, 3), self.xp, dtype)
+        a = testing.shaped_random(self.shape, self.xp, dtype)
         self.assertTrue(self.xp.all(0 <= a))
         self.assertTrue(self.xp.all(a < 10))
 
-    def test_bool(self):
-        a = testing.shaped_random(10000, self.xp, numpy.bool_)
-        self.assertTrue(4000 < self.xp.sum(a) < 6000)
-
     @testing.for_complex_dtypes()
     def test_complex(self, dtype):
-        a = testing.shaped_random((2, 3), self.xp, dtype)
+        a = testing.shaped_random(self.shape, self.xp, dtype)
         self.assertTrue(self.xp.all(0 <= a.real))
         self.assertTrue(self.xp.all(a.real < 10))
-        self.assertTrue(self.xp.all(0 <= a.imag))
-        self.assertTrue(self.xp.all(a.imag < 10))
-        self.assertTrue(self.xp.any(a.imag))
+        if 0 not in self.shape:
+            # TODO(niboshi): imag causes an error if size=0. Fix it.
+            self.assertTrue(self.xp.all(0 <= a.imag))
+            self.assertTrue(self.xp.all(a.imag < 10))
```

Comment:
Why does `0 <= a.imag < 10` fail?
`if 0 not in self.shape: self.assertTrue(self.xp.any(a.imag))` looks correct (without TODO).

Reply:
This is how the test was originally written and I'm not going to change it.
`a.imag` causes assertion error (#2885).

Reply:
Thanks!